packages/shacl/report.js

```diff
@@ -19,8 +20,7 @@ async function * validate(ds, maxViolations, iterable) {
       break
     }
 
-    // create a new validator instance at each iteration to avoid memory leaks
-    const validator = new SHACLValidator(ds, { maxErrors: 0, factory: this.env })
+    const validator = new SHACLValidator(shapes || chunk, { maxErrors: 0, factory: this.env })
```

Comment:
wondering about merging `shapes`  and `chunk`, but probably not worth it for the overhead and the added complexity

Reply:
Merging how?

Reply:
regular graph merge, hence collecting shapes from both